src/test/java/org/sigmah/server/inject/GuiceJUnitRunner.java

```diff
@@ -171,4 +214,72 @@ private static Injector createInjectorFor(final List<Class<?>> classes) throws I
 	private void startPersistUnit() {
 		injector.getInstance(PersistService.class).start();
 	}
+	
+	/**
+	 * Parse the <code>META-INF/context.xml</code> file and try to connect to 
+	 * the database.
+	 * 
+	 * @return <code>true</code> if the connection was successful, 
+	 * <code>false</code> otherwise.
+	 */
+	private boolean isDatabaseAvailable() {
+		
+		boolean available = false;
+		final SAXParserFactory factory = SAXParserFactory.newInstance();
+
+		try {
+			factory.setValidating(false);
+			factory.setFeature("http://xml.org/sax/features/validation", false);
+			factory.setFeature("http://apache.org/xml/features/nonvalidating/load-dtd-grammar", false);
+			factory.setFeature("http://apache.org/xml/features/nonvalidating/load-external-dtd", false);
+			factory.setFeature("http://xml.org/sax/features/external-general-entities", false);
+			factory.setFeature("http://xml.org/sax/features/external-parameter-entities", false);
+
+			final SAXParser parser = factory.newSAXParser();
+			
+			final String[] properties = new String[3];
+			
+			parser.parse(getClass().getResourceAsStream("/META-INF/persistence.xml"), new DefaultHandler() {
+				
+				@Override
+				public void startElement(String uri, String localName, String qName, Attributes attributes) throws SAXException {
+					if ("property".equals(qName)) {
+						final String name = attributes.getValue("name");
+						final String value = attributes.getValue("value");
+						
+						if (name != null) switch (name) {
+						case "hibernate.connection.driver_class":
+							try {
+								Class.forName(value);
+							} catch (ClassNotFoundException ex) {
+								throw new SAXException("SQL driver '" + value + "' could not be loaded.", ex);
+							}
+							break;
+						case "hibernate.connection.url":
+							properties[0] = value;
+							break;
+						case "hibernate.connection.username":
+							properties[1] = value;
+							break;
+						case "hibernate.connection.password":
+							properties[2] = value;
+							break;
+						default:
+							break;
+						}
+					}
+				}
+				
+			});
+			
+			final Connection testConnection = DriverManager.getConnection(properties[0], properties[1], properties[2]);
+			testConnection.close();
+			
+			available = true;
+		} catch (ParserConfigurationException | SAXException | IOException | SQLException ex) {
+			// Ignored.
```

Comment:
I'm not OK with that because I don't want the test to silently be ignored if there is a problem reading the persistence.xml file.

Can you split this catch clause in two catch clauses:
* `catch (SQLException ex)` should just set `available = false`
* `catch (ParserConfigurationException | SAXException | IOException ex)` should explode by throwing a RuntimeException

Also, for the reader, i would prefer that the `available = false` statement only be in the catch and the variable declared and not initialized. This way, the reader will know it's the only path for the variable to become false.

I know it may be opinion-based but please avoid empty catch clause (I forbid myself to write those, in the worst case, I log it), they sometimes lead to silently hide unexpected behavior.